src/loadr.cpp

```diff
@@ -0,0 +1,83 @@
+/* ****************************************************************************
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+*
+*
+* This file is part of Microsoft R Host.
+*
+* Microsoft R Host is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Microsoft R Host is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+*
+* ***************************************************************************/
+
+#define RHOST_NO_API_REDIRECT
+#include "stdafx.h"
+
+HMODULE r_module = nullptr;
+HMODULE rgraphapp_module = nullptr;
+
+#define _GET_PROC(m, api) _RAPI_PTR(api) = get_proc<decltype(api)>(m, _RAPI_STR(api));
+#define _RAPI_GET_PROC(api) _GET_PROC(r_module, api)
+#define _RAPI_UNLOAD(api) _RAPI_PTR(api) = nullptr;
+
+#ifdef _WIN32
+#define _RGRAPHAPPAPI_GET_PROC(api) _GET_PROC(rgraphapp_module, api)
+#endif
+
+namespace rhost {
+    namespace rapi {
+        _RAPI_SET(_RAPI_DEFINE_NULLPTR);
+        _RGRAPHAPPAPI_SET(_RAPI_DEFINE_NULLPTR);
+
+        namespace {
+            template<typename T>
+            T* get_proc(HMODULE module, std::string proc) {
```

Comment:
This should be `const std::string&`.

Although it looks like we're always actually passing string literals to in? In which case it might as well be `const char*`.